django/core/handlers/wsgi.py

```diff
@@ -159,7 +159,7 @@ def __call__(self, environ, start_response):
                     # Check that middleware is still uninitialized.
                     if self._request_middleware is None:
                         self.load_middleware()
-                except:
+                except TypeError:
```

Comment:
I tracked that code to [#11193](https://code.djangoproject.com/ticket/11193). I don't understand that ticket well enough to say for sure if the try/except is needed (I agree with your analysis unless there is some thread-safety issue that I don't understand). Unless we can reach a definitive conclusion about removing the try/except, it seems like TypeError should be Exception.